Add Cloud init

just1not2 opened the pull request:
# Short description

Add cloud-init support for the Openstack Provider


# Changes

* add cloud-init support for the Openstack Provider
* prevent crash with no keypair (OS provider)
* create a cloud-init template and gitignore its copy
* bind volumes for the template copy


# Tests

The module was launched with cloud-init

just1not2 committed:
🔧(openstack) add variables cloud-init

Add variables for the openstack cloud-init
to configure VMs at first boot.

just1not2 committed:
🎨(cloud-init) move cloud-init file

In the openstack provider, the cloud-init default
file would be copied into the docker image, which
is not good.

just1not2 committed:
✨(cloud-init) add cloud-init support

A file is opened and imported at cloud-init.

just1not2 committed:
🔖(version) release 1.1.0

Add cloud-init support.

just1not2 committed:
🔨(makefile) add cloud-init creation

In order to bind volumes correctly in the docker-compose,
the cloud-init file should be created first.

Benjou02 commented on docker-compose.yml:
```diff
@@ -9,3 +9,5 @@ services:
       - "${APP_PORT}:${APP_PORT}"
     env_file: .env
+    volumes:
+      - ./cloud-init.sh:/cloud-init.sh
```
```suggestion
      - "./cloud-init.sh:${OPENSTACK_CLOUD_INIT_FILE}"
```

Benjou02 commented on .env.template:
```diff
@@ -41,3 +41,4 @@ OPENSTACK_NETWORK=Ext-Net
 OPENSTACK_METADATA_KEY=group
 OPENSTACK_METADATA_VALUE=scaled
 OPENSTACK_KEYPAIR=
+OPENSTACK_CLOUD_INIT_FILE=
```
```suggestion
OPENSTACK_CLOUD_INIT_FILE=/cloud-init.sh
```

just1not2 committed:
🔧(docker-compose) update mount volume dir

We used the env var. Besides, as the WORKDIR is
/, it is ok to set a / prefix in order to auth-
orize relative paths.
